<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Justin Silverman" />


<title>Joint Modeling (e.g., Multiomics) with fido::Orthus</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">

div.csl-bib-body { }
div.csl-entry {
clear: both;
}
.hanging div.csl-entry {
margin-left:2em;
text-indent:-2em;
}
div.csl-left-margin {
min-width:2em;
float:left;
}
div.csl-right-inline {
margin-left:2em;
padding-left:1em;
}
div.csl-indent {
margin-left: 2em;
}
</style>

<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Joint Modeling (e.g., Multiomics) with
fido::Orthus</h1>
<h4 class="author">Justin Silverman</h4>


<div id="TOC">
<ul>
<li><a href="#fidoorthus-the-model" id="toc-fidoorthus-the-model">fido::orthus, the model</a></li>
<li><a href="#joint-modeling-of-microbial-16s-data-and-metabolomics" id="toc-joint-modeling-of-microbial-16s-data-and-metabolomics">Joint
modeling of Microbial 16S data and Metabolomics</a></li>
<li><a href="#investigate-model-results" id="toc-investigate-model-results">Investigate Model Results</a></li>
<li><a href="#qualifications-and-caution" id="toc-qualifications-and-caution">Qualifications and Caution</a></li>
<li><a href="#references" id="toc-references">References</a></li>
</ul>
</div>

<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(fido)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">48482</span>)</span></code></pre></div>
<p>If you have not already done so, I would read through the
<em>pibble</em> vignette before this one.</p>
<div id="fidoorthus-the-model" class="section level1">
<h1>fido::orthus, the model</h1>
<p>fido can be used for jointly modeling multivariate count data and
multivariate Gaussian data. For example, this would be a reasonable
model to jointly model 16S microbiome data and metabolomics data
jointly. Because of the “two-headed” nature of this model, e.g., two
observed data-sets, I named this model <em>orthus</em>, <a href="https://en.Wikipedia.org/wiki/Orthrus">a two-headed dog and
brother of Cerberus in Greek Mythology</a>. The <em>orthus</em> model
can be written as</p>
<p><span class="math display">\[
\begin{align}
Y_j &amp; \sim \text{Multinomial}\left(\pi_j \right)  \\
\pi_j &amp; = \phi^{-1}(\eta_j) \\
\begin{bmatrix}\eta_j \\ Z_j \end{bmatrix} &amp;\sim N(\Lambda X,
\Sigma) \\
\Lambda &amp;\sim  N(\Theta, \Sigma, \Gamma) \\
\Sigma &amp;\sim W^{-1}(\Xi, \upsilon)
\end{align}
\]</span></p>
<p>Note this looks nearly identical to the <em>pibble</em> model but we
have appended the second (Gaussian) dataset (<span class="math inline">\(Z\)</span>) onto <span class="math inline">\(\eta\)</span>. In doing this, the definition of
<span class="math inline">\(\Lambda\)</span> changes (it is now larger
with the bottom rows dictating how the covariates <span class="math inline">\(X\)</span> influence the second dataset).
Similarly, <span class="math inline">\(\Sigma\)</span> now is much
larger and can be though of as <span class="math display">\[
\Sigma = \begin{bmatrix} \Sigma_{(\eta, \eta)} &amp; \Sigma_{(\eta, Z)}
\\
                          \Sigma_{(Z, \eta)} &amp; \Sigma_{(Z,
Z)}\end{bmatrix}
\]</span> where <span class="math inline">\(\Sigma_{(\eta,
\eta)}\)</span> describes the covariance between log-ratios (e.g., the
covariance among the multinomial categories in log-ratio space), <span class="math inline">\(\Sigma_{(Z, Z)}\)</span> describes the covariance
between the dimensions of <span class="math inline">\(Z\)</span> (e.g.,
between metabolites if Z is metabolomics data), and <span class="math inline">\(\Sigma_{(\eta, Z)} = \Sigma_{(Z, \eta)}^T\)</span>
represents the covariance between log-ratios and dimensions of <span class="math inline">\(Z\)</span> (e.g., between microbial taxa and
metabolites). Similar to <span class="math inline">\(\Sigma\)</span> and
<span class="math inline">\(\Lambda\)</span>, the parameters <span class="math inline">\(\Xi\)</span> and <span class="math inline">\(\Theta\)</span> undergo a similar expansion to
accommodate the second dataset.</p>
</div>
<div id="joint-modeling-of-microbial-16s-data-and-metabolomics" class="section level1">
<h1>Joint modeling of Microbial 16S data and Metabolomics</h1>
<p>To demonstrate <em>orthus</em> I will perform a toy analysis on data
from <span class="citation">Kashyap et al. (2013)</span> and made
available by <span class="citation">Callahan et al. (2016)</span> as
part of their recently published microbiome data analysis workflow <span class="citation">(Callahan et al. 2016)</span>. I follow the data
preprocessing of <span class="citation">Callahan et al. (2016)</span> we
just don’t drop taxa but instead amalgamate those that don’t pass
filtering to a category called “other”. I do this to maintain the proper
variance in the multinomial model.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>metab_path <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata/Kashyap2013&quot;</span>, <span class="st">&quot;metabolites.csv&quot;</span>, <span class="at">package=</span><span class="st">&quot;fido&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>microbe_path <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata/Kashyap2013&quot;</span>, <span class="st">&quot;microbe.rda&quot;</span>, <span class="at">package=</span><span class="st">&quot;fido&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>metab <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(metab_path, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>metab <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(metab)</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>microbe <span class="ot">&lt;-</span> <span class="fu">get</span>(<span class="fu">load</span>(microbe_path))</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="do">## Preprocessing ##</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co"># Metabolite Preprocessing</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>keep_ix <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(metab <span class="sc">==</span> <span class="dv">0</span>) <span class="sc">&lt;=</span> <span class="dv">3</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>metab <span class="ot">&lt;-</span> metab[keep_ix, ]</span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="co"># 16S Preprocesing - plus some weirdness to rename amalgamated category to &quot;other&quot;</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>keep_ix <span class="ot">&lt;-</span> <span class="fu">taxa_sums</span>(microbe) <span class="sc">&gt;</span> <span class="dv">4</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>keep_ix <span class="ot">&lt;-</span> keep_ix <span class="sc">&amp;</span> (<span class="fu">rowSums</span>(<span class="fu">otu_table</span>(microbe)<span class="sc">&gt;</span><span class="dv">2</span>)<span class="sc">&gt;</span><span class="dv">3</span>)</span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a>microbe <span class="ot">&lt;-</span> <span class="fu">merge_taxa</span>(microbe, <span class="fu">taxa_names</span>(microbe)[<span class="sc">!</span>keep_ix])</span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a>nms <span class="ot">&lt;-</span> <span class="fu">taxa_names</span>(microbe)</span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a>rnm <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">taxa_names</span>(microbe)<span class="sc">==</span><span class="fu">taxa_names</span>(microbe)[<span class="sc">!</span>keep_ix][<span class="dv">1</span>])</span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a>nms[rnm] <span class="ot">&lt;-</span> <span class="st">&quot;other&quot;</span></span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="fu">taxa_names</span>(microbe) <span class="ot">&lt;-</span> nms</span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="fu">rm</span>(nms, rnm)</span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a></span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a><span class="co"># bit of preprocessing </span></span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a>metab <span class="ot">&lt;-</span> <span class="fu">log10</span>(<span class="dv">1</span> <span class="sc">+</span> metab)</span></code></pre></div>
<p>Now I am going to do just a bit of processing to get data into a
format for orthus. Note I have no extra metadata so we are just going to
use an intercept in our model at this time.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>Y <span class="ot">&lt;-</span> <span class="fu">otu_table</span>(microbe, <span class="at">taxa_are_rows=</span><span class="cn">TRUE</span>)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>Z <span class="ot">&lt;-</span> metab <span class="co">#(metabolites are rows)</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>X <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">1</span>, <span class="dv">1</span>, phyloseq<span class="sc">::</span><span class="fu">nsamples</span>(microbe))</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co"># save dims for easy reference</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>N <span class="ot">&lt;-</span> <span class="fu">ncol</span>(Y)</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>P <span class="ot">&lt;-</span> <span class="fu">nrow</span>(Z)</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>Q <span class="ot">&lt;-</span> <span class="fu">nrow</span>(X)</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>D <span class="ot">&lt;-</span> <span class="fu">nrow</span>(Y)</span></code></pre></div>
<p>Now I am going to set up the priors. My priors are going to be
similar to that of <em>pibble</em> but now we need to think about a
prior for the covariance among the metabolites and between the
metabolites and the log-ratios of the taxa. Remember, that priors must
be defined in the <span class="math inline">\(ALR_D\)</span> (e.g., ALR
with the reference being the D-th taxa; this may be changed in the
future to make specifying priors more user friendly).</p>
<p>I am going to form our prior for <span class="math inline">\(\Sigma\)</span> by specifying <span class="math inline">\(\upsilon\)</span> and <span class="math inline">\(\Xi\)</span>. I will specify that I have weak
prior belief that the taxa are independent in terms of their log
absolute abundance. We can translate this statement about covariance of
log absolute abundance into a statement about log-ratio covariance by
pre- and post-multiplying by the <span class="math inline">\(ALR_D\)</span> contrast matrix (which I refer to
as <span class="math inline">\(GG\)</span> below). Additionally, I
believe that there is likely no substantial covariance between the taxa
and the metabolites and I assume the metabolites are likely
independent.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>upsilon <span class="ot">&lt;-</span> (D<span class="dv">-1</span><span class="sc">+</span>P)<span class="sc">+</span><span class="dv">10</span> <span class="co"># weak-ish prior on covariance over joint taxa and metabolites</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>Xi <span class="ot">&lt;-</span> <span class="fu">diag</span>(D<span class="dv">-1</span><span class="sc">+</span>P)</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>GG <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">diag</span>(D<span class="dv">-1</span>), <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>Xi[<span class="dv">1</span><span class="sc">:</span>(D<span class="dv">-1</span>), <span class="dv">1</span><span class="sc">:</span>(D<span class="dv">-1</span>)] <span class="ot">&lt;-</span> GG<span class="sc">%*%</span><span class="fu">diag</span>(D) <span class="sc">%*%</span> <span class="fu">t</span>(GG)</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>Xi <span class="ot">&lt;-</span> Xi <span class="sc">*</span> (upsilon<span class="sc">-</span>D<span class="sc">-</span>P) <span class="co"># this scales Xi to have the proper mean we wanted</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="fu">image</span>(Xi)</span></code></pre></div>
<p><img src="https://github.com/jsilve24/fido/raw/master/vignettes/orthus-prior-structure.png" width="50%" style="display: block; margin: auto;" /></p>
<p>Note the structure of this prior, everything is independent but there
is a moderate positive covariance between the log-ratios based on their
shared definition in terms of the <span class="math inline">\(D\)</span>-th taxa.</p>
<p>The other parts of the prior are less interesting. We are going to
state that our mean for <span class="math inline">\(\Lambda\)</span> is
centered about <span class="math inline">\(\mathbf{0}\)</span> and that
the signal-to-noise ratio in the data is approximately 1 (this later
part is specified by <span class="math inline">\(\Gamma=I\)</span>).</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>Gamma <span class="ot">&lt;-</span> <span class="fu">diag</span>(Q)</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>Theta <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, D<span class="dv">-1</span><span class="sc">+</span>P, Q)</span></code></pre></div>
<p>Finally I fit the model.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">orthus</span>(Y, Z, X, <span class="at">Theta=</span>Theta, <span class="at">Gamma=</span>Gamma, <span class="at">Xi=</span>Xi, <span class="at">upsilon=</span>upsilon, <span class="at">n_samples=</span><span class="dv">1000</span>)</span></code></pre></div>
<p>Next we are going to transform the log-ratios from <span class="math inline">\(ALR_D\)</span> to the <span class="math inline">\(CLR\)</span>. I have written all the
transformation functions, <em>e.g.</em>, <code>to_clr</code> etc… to
work on <code>orthusfit</code> objects in a similar manner to how they
work on <code>pibblefit</code> objects. For <code>orthusfit</code>
objects they only transform the log-ratio components of parameters
leaving the other parts of inferred model parameters (<em>i.e.</em>, the
parts associated with the metabolites) untouched.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">to_clr</span>(fit)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a><span class="fu">print</span>(fit)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="co">#&gt; orthusfit Object: </span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co">#&gt;   Number of Samples:      12 </span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co">#&gt;   Number of Categories:       114 </span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co">#&gt;   Number of Zdimensions:  405 </span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="co">#&gt;   Number of Covariates:       1 </span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="co">#&gt;   Number of Posterior Samples:    1000 </span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="co">#&gt;   Contains Samples of Parameters:Eta  Lambda  Sigma</span></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="co">#&gt;   Coordinate System:      clr</span></span></code></pre></div>
</div>
<div id="investigate-model-results" class="section level1">
<h1>Investigate Model Results</h1>
<p>There are a ton of ways to visualize the inferred model. I could make
network diagrams relating taxa to taxa, taxa to metabolites and
metabolites to metabolites. I could look at a low dimensional
representation of joint covariance to create something very much akin to
canonical correlation analysis (CCA). I could look at how well the
metabolites predict the taxa and vice-versa. But for the sake of
simplicity I will do something much simpler. Here I am just going to
find a list of taxa metabolite covariances that the model is very
confident about.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co"># First just look ath the cross-covariances fit by the model</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="co"># (covariance between taxa in CLR coordinates and metabolites)</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co"># This requires that we extract the corner of Sigma. </span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>xcor <span class="ot">&lt;-</span> fit<span class="sc">$</span>Sigma[<span class="dv">1</span><span class="sc">:</span>D, D<span class="sc">:</span>(D<span class="dv">-1</span><span class="sc">+</span>P),]</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co"># Initial preprocessing to speed up computation of posterior intervals</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="co"># As there are a lot of cross-covariance terms we are going to first </span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="co"># weed down the list of things we have to look at by first pass </span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="co"># selecting only those taxa that have a large posterior mean for the covariance</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>xcor.mean <span class="ot">&lt;-</span> <span class="fu">apply</span>(xcor, <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), mean)</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>to.analyze <span class="ot">&lt;-</span> fido<span class="sc">::</span><span class="fu">gather_array</span>(xcor.mean, cov, taxa, metabolite) <span class="sc">%&gt;%</span> </span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span><span class="fu">abs</span>(cov)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>  .[<span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>,] <span class="sc">%&gt;%</span> </span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">tm =</span><span class="fu">paste0</span>(taxa, <span class="st">&quot;_&quot;</span>, metabolite))</span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a><span class="co"># Subset Covariance to those we are interested in and calculate posterior </span></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a><span class="co"># confidence intervals. </span></span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a>xcor.summary <span class="ot">&lt;-</span> fido<span class="sc">::</span><span class="fu">gather_array</span>(xcor, cov, taxa, metabolite, iter) <span class="sc">%&gt;%</span></span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">tm=</span><span class="fu">paste0</span>(taxa, <span class="st">&quot;_&quot;</span>, metabolite)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a>  <span class="fu">filter</span>(tm <span class="sc">%in%</span> to.analyze<span class="sc">$</span>tm) <span class="sc">%&gt;%</span> </span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">taxa =</span> <span class="fu">rownames</span>(Y)[taxa], <span class="at">metabolite =</span> <span class="fu">rownames</span>(Z)[metabolite]) <span class="sc">%&gt;%</span> </span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a>  <span class="fu">group_by</span>(taxa, metabolite) <span class="sc">%&gt;%</span> </span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a>  fido<span class="sc">:::</span><span class="fu">summarise_posterior</span>(cov) <span class="sc">%&gt;%</span> </span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a>  <span class="fu">arrange</span>(mean) <span class="sc">%&gt;%</span> </span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a>  <span class="fu">filter</span>(taxa <span class="sc">!=</span> <span class="st">&#39;other&#39;</span>) <span class="co"># we don&#39;t care about these</span></span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a></span>
<span id="cb8-27"><a href="#cb8-27" tabindex="-1"></a><span class="co"># Select those covariances where the model has high certainty (95%) that</span></span>
<span id="cb8-28"><a href="#cb8-28" tabindex="-1"></a><span class="co"># the true covariance is not zero. </span></span>
<span id="cb8-29"><a href="#cb8-29" tabindex="-1"></a>xcor.summary <span class="sc">%&gt;%</span> </span>
<span id="cb8-30"><a href="#cb8-30" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">sign</span>(p2<span class="fl">.5</span>)<span class="sc">==</span><span class="fu">sign</span>(p97<span class="fl">.5</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-31"><a href="#cb8-31" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">abs</span>(mean) <span class="sc">&gt;</span> <span class="dv">2</span>)</span>
<span id="cb8-32"><a href="#cb8-32" tabindex="-1"></a><span class="co">#&gt; # A tibble: 218 x 8</span></span>
<span id="cb8-33"><a href="#cb8-33" tabindex="-1"></a><span class="co">#&gt; # Groups:   taxa [17]</span></span>
<span id="cb8-34"><a href="#cb8-34" tabindex="-1"></a><span class="co">#&gt;    taxa  metabolite   p2.5   p25   p50  mean   p75  p97.5</span></span>
<span id="cb8-35"><a href="#cb8-35" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;</span></span>
<span id="cb8-36"><a href="#cb8-36" tabindex="-1"></a><span class="co">#&gt;  1 722   206.0445922 -6.55 -3.96 -3.08 -3.33 -2.32 -1.51 </span></span>
<span id="cb8-37"><a href="#cb8-37" tabindex="-1"></a><span class="co">#&gt;  2 7816  206.0445922 -5.63 -3.23 -2.45 -2.65 -1.85 -1.05 </span></span>
<span id="cb8-38"><a href="#cb8-38" tabindex="-1"></a><span class="co">#&gt;  3 722   290.9298419 -5.18 -3.09 -2.42 -2.62 -1.85 -1.23 </span></span>
<span id="cb8-39"><a href="#cb8-39" tabindex="-1"></a><span class="co">#&gt;  4 18182 380.1846197 -5.20 -3.07 -2.38 -2.55 -1.83 -1.03 </span></span>
<span id="cb8-40"><a href="#cb8-40" tabindex="-1"></a><span class="co">#&gt;  5 722   181.4504354 -5.19 -3.04 -2.36 -2.55 -1.80 -1.13 </span></span>
<span id="cb8-41"><a href="#cb8-41" tabindex="-1"></a><span class="co">#&gt;  6 722   177.0565368 -4.98 -3.01 -2.30 -2.50 -1.78 -1.08 </span></span>
<span id="cb8-42"><a href="#cb8-42" tabindex="-1"></a><span class="co">#&gt;  7 722   180.072273  -5.03 -3.06 -2.30 -2.49 -1.71 -0.986</span></span>
<span id="cb8-43"><a href="#cb8-43" tabindex="-1"></a><span class="co">#&gt;  8 19517 380.1846197 -5.15 -3.06 -2.33 -2.49 -1.71 -0.952</span></span>
<span id="cb8-44"><a href="#cb8-44" tabindex="-1"></a><span class="co">#&gt;  9 2943  380.1846197 -4.89 -2.97 -2.34 -2.49 -1.83 -1.03 </span></span>
<span id="cb8-45"><a href="#cb8-45" tabindex="-1"></a><span class="co">#&gt; 10 722   176.0343919 -4.93 -2.95 -2.25 -2.48 -1.79 -1.09 </span></span>
<span id="cb8-46"><a href="#cb8-46" tabindex="-1"></a><span class="co">#&gt; # ... with 208 more rows</span></span></code></pre></div>
<p>So it looks there there are a few hundred covariances that we can be
fairly confident about.</p>
</div>
<div id="qualifications-and-caution" class="section level1">
<h1>Qualifications and Caution</h1>
<p>Please note, I performed this analysis to demonstrate the use of
<em>orthus</em> which is a model that I have been repeatedly asked for.
I think its a cool model and could be quite useful in the right
circumstances. But I would like to point out a few philosophical points
about the analysis I performed above.</p>
<p>First, I performed this analysis just to demonstrate <em>orthus</em>.
I really don’t know the data showcased here. What is metabolite
<code>206.0445922</code>? I have no idea. For some reason this is how
the metabolites in that dataset were named. For the same reason I have
left the taxa indexed by sequence variant number.</p>
<p>Second (and more important), identifying relationships between taxa
and metabolites (or between any two high-dimensional multivariate
data-sets) is really difficult! Here we are looking at just 114 taxa and
405 but this leads to 46170 possible covariances and here we only have
12 samples! Yes <em>orthus</em> is a Bayesian model, and Yes, Bayesian
models can be quite useful when there are more parameters than samples,
but there is a limit of reasonability. Really, Bayesian models are great
when you can perfectly capture your prior beliefs with your prior. But
how often can that really be done perfectly? As such I would caution
users, to use <em>orthus</em> carefully. Consider which metabolites and
taxa you really care about and if you can, isolate your analyses to
those.</p>
<p>Alright, that’s probably enough philosophizing for an R package
Vignette. I hope you enjoy <em>orthus</em>.</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-callahan2016" class="csl-entry">
Callahan, Ben J, Kris Sankaran, Julia A Fukuyama, Paul J McMurdie, and
Susan P Holmes. 2016. <span>“Bioconductor Workflow for Microbiome Data
Analysis: From Raw Reads to Community Analyses.”</span>
<em>F1000Research</em> 5.
</div>
<div id="ref-kashyap2013" class="csl-entry">
Kashyap, Purna C, Angela Marcobal, Luke K Ursell, Samuel A Smits, Erica
D Sonnenburg, Elizabeth K Costello, Steven K Higginbottom, et al. 2013.
<span>“Genetically Dictated Change in Host Mucus Carbohydrate Landscape
Exerts a Diet-Dependent Effect on the Gut Microbiota.”</span>
<em>Proceedings of the National Academy of Sciences</em> 110 (42):
17059–64.
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
